String/test/CheckExceeding.test.js

```diff
@@ -0,0 +1,38 @@
+import { checkExceeding } from '../CheckExceeding'
+
+describe('Testing CheckExceeding function', () => {
+  it('Testing the invalid types', () => {
+    expect(() => checkExceeding(Math.random())).toThrow('Argument is not a string')
+    expect(() => checkExceeding(null)).toThrow('Argument is not a string')
+    expect(() => checkExceeding(false)).toThrow('Argument is not a string')
+    expect(() => checkExceeding(false)).toThrow('Argument is not a string')
+  })
+
+  it('Testing with empty string', () => {
+    expect(checkExceeding('')).toBe(true)
+  })
+
+  it('Testing with linear alphabets', () => {
+    expect(checkExceeding('a b c d e ')).toBe(true)
+    expect(checkExceeding('f g h i j ')).toBe(true)
+    expect(checkExceeding('k l m n o ')).toBe(true)
+    expect(checkExceeding('p q r s t ')).toBe(true)
+    expect(checkExceeding('u v w x y z')).toBe(true)
+  })
+
+  it('Testing not exceeding words', () => {
+    expect(checkExceeding('Hello')).toBe(false)
+    expect(checkExceeding('world')).toBe(false)
+    expect(checkExceeding('update')).toBe(false)
+    expect(checkExceeding('upgrade')).toBe(false)
+  })
+
+  it('Testing exceeding words', () => {
+    expect(checkExceeding('bee')).toBe(true) // [ 3 ]
+    expect(checkExceeding('can')).toBe(true) // [ 2, 13 ]
```

Comment:
Why is `can` exceeding? c > a < n

Reply:
By the rules of the **lexicographical system**, your statement is 100% true. but this solution does not follow the lexicographical system.

Reply:
More context?

Reply:
If I add a lexicographical condition, It'll be hard to find the increasing words. Only linear alphabets exist in the exceeding word list, which has no meaning and have very tiny examples.

Reply:
If I add **lexicographical** condition, it's hard to find the **increments** words.

Reply:
No, I wanted to know what definition of an exceeding word are you using.

Reply:
From the problem, the definition is, every **ASCII** difference of **char** & **adjacentChar** sequence are must be increasing.

Reply:
Yes, but if you do a `Math.Abs()`, all of them will be positive, so increasing, so I don't get how this works.

Reply:
I used `Math.abs()` for the "update" example that increasing `[5, 12, 3, 19, 15]`, not `[ 5, 12, 3, -19, 15 ]`

Reply:
I still do not understand what you are trying to do. I will try to make sense of the algorithm again tomorrow, but in the mean type, provide an explanation of how the algorithm works.

Reply:
From the challenge 
> Exceeding words are words where the gap between two adjacent characters is increasing.
**The gap is the distance in ASCII**

Example: **"update** is not exceeding, because the ASCII distance of each pair isn't increasing.

Explanation of **update** word: 
from `u:117` to `p:112` ->  `117 - 112 = |5| = 5`
from `p:112` to `d:100` -> `112 - 100 = |12| = 12`
from `d:100` to `a:97` -> `100 - 97 = |3| = 3`
from `a:97` to `t:116`: -> `97 - 116 = |-19| = 19`
from `t:116` to `e:101`: `116 - 101 = |15| = 15`

From this explanation, we can find `[5, 12, 3, 19. 15]`  that is not increasing. This is why the **update** is not exceeding. If we found any increasing sequence then it'll be exceeding. that's it.

Reply:
Can you give an example of an exceeding word and explain why it is exceeding?

Reply:
Sure, let's see with the "delete" word, It's exceeding because the **ASCII** distance of each pair is increasing.

Explanation of "**delete**" word:
from `d:117` to `e:101`-> `100- 101= |-1| = 1`
from `e:101`to `l:108`-> `101- 108= |-7| = 7`
from `l:108` to` e:101` -> `108 - 101= |7| = 7`
from `e:101` to `t:116` -> `101 - 116 = |-15| = 15`
from `t:116` to `e:101` -> `116 - 101 = |15| = 15`

From this, we got `[1, 7, 7, 15, 15]` which is increasing. that's why It should call exceeding.

Reply:
I get the misunderstanding we were having now. I will try to resolve it a bit later 👍🏽

Reply:
@raklaptudirm, Will you resolve this?